test/common/IotHub.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace common
+{
+    using System;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Common;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Azure.EventHubs;
+    using Newtonsoft.Json;
+
+    using DeviceTransportType = Microsoft.Azure.Devices.TransportType;
+    using EventHubTransportType = Microsoft.Azure.EventHubs.TransportType;
+
+    public class IotHub
+    {
+        readonly string eventHubEndpoint;
+        readonly string iotHubConnectionString;
+        readonly Lazy<RegistryManager> registryManager;
+        readonly Lazy<ServiceClient> serviceClient;
+        readonly Lazy<EventHubClient> eventHubClient;
+
+        public string Hostname =>
+            IotHubConnectionStringBuilder.Create(this.iotHubConnectionString).HostName;
+        public string EntityPath =>
+            (new EventHubsConnectionStringBuilder(this.eventHubEndpoint)).EntityPath;
+
+        RegistryManager RegistryManager => this.registryManager.Value;
+        ServiceClient ServiceClient => this.serviceClient.Value;
+        EventHubClient EventHubClient => this.eventHubClient.Value;
+
+        public IotHub(string iotHubConnectionString, string eventHubEndpoint, Option<string> proxyUrl)
+        {
+            this.eventHubEndpoint = eventHubEndpoint;
+            this.iotHubConnectionString = iotHubConnectionString;
+            Option<IWebProxy> proxy = proxyUrl.Map(p => new WebProxy(p) as IWebProxy);
+
+            this.registryManager = new Lazy<RegistryManager>(() =>
+            {
+                var settings = new HttpTransportSettings();
+                proxy.ForEach(p => settings.Proxy = p);
+                return RegistryManager.CreateFromConnectionString(
+                    this.iotHubConnectionString,
+                    settings
+                );
+            });
+
+            this.serviceClient = new Lazy<ServiceClient>(() =>
+            {
+                var settings = new ServiceClientTransportSettings();
+                proxy.ForEach(p => settings.HttpProxy = p);
+                return ServiceClient.CreateFromConnectionString(
+                    this.iotHubConnectionString,
+                    DeviceTransportType.Amqp_WebSocket_Only,
```

Comment:
why use AmqpWs as default device transport type?

Reply:
Why not? That way it will always work (proxy or non-proxy environment). The protocol used by the service client a part of what's being tested, so I chose the simplest route.

Reply:
sure.  it can be configurable in the future. Direct Method Cloud Sender module needs to support different protocol types.